### PR TITLE
Validate constituency from postcode input

### LINKED ISSUE #443: Show error if postcode is outside the two constituencies
If a postcode outside either of these constituencies is entered, it should display a suitable error.

### PR DESCRIPTION
**Pull request by aspiers**
Partially fixes #443, but needs to be prettier and do backend validation too.

Could switch to using https://jqueryvalidation.org/jQuery.validator.addMethod/ for form validation.

### CODE REVIEW

**Comment by aspiers** [comment]
@baob How are your Javascript skills ;-)  This is just a sketch, not a complete solution.

File: app/javascript/packs/postcodesHelper.js [hunk 1]
```diff
@@ -32,21 +15,46 @@ $(document).ready(() => {
     }
   };
 
-  const ajaxSuccess = () => {
-    $("#error-postcode").html("");
+  const handlePostcodeSubmit = () => {
+    var input = $("#txt-postcode").val();
+    var url = "https://api.postcodes.io/postcodes/" + input;
```
Comment (baob) [inline app/javascript/packs/postcodesHelper.js:20]: I think I would make these two vars into consts,

File: app/javascript/packs/postcodesHelper.js [hunk 2]
```diff
@@ -32,21 +15,46 @@ $(document).ready(() => {
...
-    return $.ajax({
-      url: url,
-      success: ajaxSuccess,
-      error: ajaxError
-    });
+  const handlePostcodeLookup = postcode => {
+    var onsId = postcode.result.codes.parliamentary_constituency;
+    var name = postcode.result.parliamentary_constituency;
+
+    var hasOption = $('select#user_constituency_ons_id option[value="' + onsId + '"]');
```
Comment (baob) [inline app/javascript/packs/postcodesHelper.js:36]: I think I would make these 3 vars into consts

**Review by baob — APPROVED** [review]

File: app/javascript/packs/postcodesHelper.js (hunk 1 above)
Comment (aspiers) [reply app/javascript/packs/postcodesHelper.js]: Done

File: app/javascript/packs/postcodesHelper.js (hunk 2 above)
Comment (aspiers) [reply app/javascript/packs/postcodesHelper.js]: Done

**Pull request merged**